shared/src/main/kotlin/com/egm/stellio/shared/model/CompactedEntity.kt

```diff
@@ -23,17 +27,33 @@ private fun simplifyRepresentation(value: Any): Any =
         // an attribute with a single instance
         is Map<*, *> -> simplifyValue(value as Map<String, Any>)
         // an attribute with multiple instances
-        is List<*> -> value.map {
-            when (it) {
-                is Map<*, *> -> simplifyValue(it as Map<String, Any>)
+        is List<*> -> {
+            when (value.first()) {
+                is Map<*, *> -> simplifyMultiInstanceAttribute(value as List<Map<String, Any>>)
                 // we keep @context value as it is (List<String>)
-                else -> it
+                else -> value
             }
         }
         // keep id, type and other non-reified properties as they are (typically string or list)
         else -> value
     }
 
+private fun simplifyMultiInstanceAttribute(value: List<Map<String, Any>>): Map<String, Map<String, Any>> {
+    val datasetIds = value.map {
+        val datasetId = (it[NGSILD_DATASET_ID_TERM] as? String) ?: NGSILD_NONE_TERM
+        val attributeType = it[JSONLD_TYPE_TERM] as String
+        val attributeEnum: AttributeType = AttributeType.valueOf(attributeType.uppercase(Locale.getDefault()))
+        val datasetValue: Any = when (attributeEnum) {
+            AttributeType.PROPERTY -> it[JSONLD_VALUE_TERM] as Any
+            AttributeType.RELATIONSHIP -> it[JSONLD_OBJECT] as Any
+            AttributeType.GEOPROPERTY -> it[JSONLD_VALUE_TERM] as Any
+            AttributeType.JSONPROPERTY -> it[JSONLD_JSON_TERM] as Any
+        }
```

Comment:
why not using the already existing `simplifyValue` function below? (this is what you are doing / should do for each value in a multi-attribute instance)